src/bootstrap/test.rs

```diff
@@ -1082,10 +1082,8 @@ impl Step for Compiletest {
             if builder.config.rust_optimize_tests {
                 flags.push("-O".to_string());
             }
-            if builder.config.rust_debuginfo_tests {
-                flags.push("-g".to_string());
-            }
         }
+        flags.push(format!("-C debuginfo={}", builder.config.rust_debuginfo_level_tests));
```

Comment:
I'd drop the space here just in case -- I think we sometimes parse the flags a bit and this might cause problems.